### DISCUSSION TITLE
NordVPN Issues

### DISCUSSION THREAD

**Discussion by SlippingEels** [haugene/docker-transmission-openvpn#2680] [General]
Hello,

I have an issue where the container reboots. Any advice would be great.

Config
PUID=1000
PGID=100
OPENVPN_PROVIDER=NORDVPN
OPENVPN_USERNAME=xxx
OPENVPN_PASSWORD=xxx
LOCAL_NETWORK=192.168.1.0/24
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
GLOBAL_APPLY_PERMISSIONS=true
TRANSMISSION_HOME=/config/transmission-home
TRANSMISSION_RPC_PORT=9091
TRANSMISSION_RPC_USERNAME=
TRANSMISSION_RPC_PASSWORD=
TRANSMISSION_DOWNLOAD_DIR=/data/completed
TRANSMISSION_INCOMPLETE_DIR=/data/incomplete
TRANSMISSION_WATCH_DIR=/data/watch
CREATE_TUN_DEVICE=true
ENABLE_UFW=false
UFW_ALLOW_GW_NET=false
UFW_EXTRA_PORTS=
UFW_DISABLE_IPTABLES_REJECT=false
PEER_DNS=true
PEER_DNS_PIN_ROUTES=true
DROP_DEFAULT_ROUTE=
WEBPROXY_ENABLED=false
WEBPROXY_PORT=8118
WEBPROXY_USERNAME=
WEBPROXY_PASSWORD=
LOG_TO_STDOUT=false
HEALTH_CHECK_HOST=google.com
SELFHEAL=false
REVISION=1103172c3288b7de681e2fb7f1378314f17f66cf
NORDVPN_CATEGORY=legacy_p2p
NORDVPN_PROTOCOL=tcp

Logs

Starting container with revision: 1103172c3288b7de681e2fb7f1378314f17f66cf
TRANSMISSION_HOME is currently set to: /config/transmission-home
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
/etc/openvpn/nordvpn/..
INFO: OVPN: Checking curl installation
INFO: OVPN: DNS resolution ok
INFO: OVPN: ok, configurations download site reachable
INFO: OVPN: Removing existing configs in /etc/openvpn/nordvpn
Checking NORDPVN API responses
INFO: OVPN:Selecting the best server...
WARNING: OVPN: empty or invalid NORDVPN_COUNTRY (value=). Ignoring this parameter. Possible values are:AL,AR,AU,AT,BE,BA,BR,BG,CA,CL,CO,CR,HR,CY,CZ,DK,EE,FI,FR,GE,DE,GR,HK,HU,IS,ID,IE,IL,IT,JP,LV,LT,LU,MY,MX,MD,NL,NZ,MK,NO,PL,PT,RO,RS,SG,SK,SI,ZA,KR,ES,SE,CH,TW,TH,TR,UA,AE,GB,US,VN, or Albania,Argentina,Australia,Austria,Belgium,Bosnia and Herzegovina,Brazil,Bulgaria,Canada,Chile,Colombia,Costa Rica,Croatia,Cyprus,Czech Republic,Denmark,Estonia,Finland,France,Georgia,Germany,Greece,Hong Kong,Hungary,Iceland,Indonesia,Ireland,Israel,Italy,Japan,Latvia,Lithuania,Luxembourg,Malaysia,Mexico,Moldova,Netherlands,New Zealand,North Macedonia,Norway,Poland,Portugal,Romania,Serbia,Singapore,Slovakia,Slovenia,South Africa,South Korea,Spain,Sweden,Switzerland,Taiwan,Thailand,Turkey,Ukraine,United Arab Emirates,United Kingdom,United States,Vietnam,. Please check https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn
INFO: OVPN: Searching for group: legacy_p2p
INFO: OVPN:Searching for technology: openvpn_tcp
INFO: OVPN: Best server : uk2305.nordvpn.com, load: 15
Best server : uk2305.nordvpn.com
INFO: OVPN: Downloading config: uk2305.nordvpn.com.ovpn
INFO: OVPN: Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/uk2305.nordvpn.com.tcp.ovpn
OVPN: NORDVPN: selected: uk2305.nordvpn.com, VPN_PROVIDER_HOME: /etc/openvpn/nordvpn
Starting OpenVPN using config uk2305.nordvpn.com.ovpn
Modifying /etc/openvpn/nordvpn/uk2305.nordvpn.com.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0
RTNETLINK answers: Operation not permitted
Starting container with revision: 1103172c3288b7de681e2fb7f1378314f17f66cf
TRANSMISSION_HOME is currently set to: /config/transmission-home
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
/etc/openvpn/nordvpn/..
INFO: OVPN: Checking curl installation
INFO: OVPN: DNS resolution ok
INFO: OVPN: ok, configurations download site reachable
INFO: OVPN: Removing existing configs in /etc/openvpn/nordvpn
Checking NORDPVN API responses
INFO: OVPN:Selecting the best server...
WARNING: OVPN: empty or invalid NORDVPN_COUNTRY (value=). Ignoring this parameter. Possible values are:AL,AR,AU,AT,BE,BA,BR,BG,CA,CL,CO,CR,HR,CY,CZ,DK,EE,FI,FR,GE,DE,GR,HK,HU,IS,ID,IE,IL,IT,JP,LV,LT,LU,MY,MX,MD,NL,NZ,MK,NO,PL,PT,RO,RS,SG,SK,SI,ZA,KR,ES,SE,CH,TW,TH,TR,UA,AE,GB,US,VN, or Albania,Argentina,Australia,Austria,Belgium,Bosnia and Herzegovina,Brazil,Bulgaria,Canada,Chile,Colombia,Costa Rica,Croatia,Cyprus,Czech Republic,Denmark,Estonia,Finland,France,Georgia,Germany,Greece,Hong Kong,Hungary,Iceland,Indonesia,Ireland,Israel,Italy,Japan,Latvia,Lithuania,Luxembourg,Malaysia,Mexico,Moldova,Netherlands,New Zealand,North Macedonia,Norway,Poland,Portugal,Romania,Serbia,Singapore,Slovakia,Slovenia,South Africa,South Korea,Spain,Sweden,Switzerland,Taiwan,Thailand,Turkey,Ukraine,United Arab Emirates,United Kingdom,United States,Vietnam,. Please check https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn
INFO: OVPN: Searching for group: legacy_p2p
INFO: OVPN:Searching for technology: openvpn_tcp
INFO: OVPN: Best server : uk2305.nordvpn.com, load: 15
Best server : uk2305.nordvpn.com
INFO: OVPN: Downloading config: uk2305.nordvpn.com.ovpn
INFO: OVPN: Downloading from: https://downloads.nordcdn.com/configs/files/ovpn_tcp/servers/uk2305.nordvpn.com.tcp.ovpn
OVPN: NORDVPN: selected: uk2305.nordvpn.com, VPN_PROVIDER_HOME: /etc/openvpn/nordvpn
Starting OpenVPN using config uk2305.nordvpn.com.ovpn
Modifying /etc/openvpn/nordvpn/uk2305.nordvpn.com.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0
RTNETLINK answers: Operation not permitted
Starting container with revision: 1103172c3288b7de681e2fb7f1378314f17f66cf
TRANSMISSION_HOME is currently set to: /config/transmission-home
Creating TUN device /dev/net/tun
Using OpenVPN provider: NORDVPN
Running with VPN_CONFIG_SOURCE auto
Provider NORDVPN has a bundled setup script. Defaulting to internal config
Executing setup script for NORDVPN
/etc/openvpn/nordvpn/..
INFO: OVPN: Checking curl installation
INFO: OVPN: DNS resolution ok
INFO: OVPN: ok, configurations download site reachable
INFO: OVPN: Removing existing configs in /etc/openvpn/nordvpn
Checking NORDPVN API responses
INFO: OVPN:Selecting the best server...
WARNING: OVPN: empty or invalid NORDVPN_COUNTRY (value=). Ignoring this parameter. Possible values are:AL,AR,AU,AT,BE,BA,BR,BG,CA,CL,CO,CR,HR,CY,CZ,DK,EE,FI,FR,GE,DE,GR,HK,HU,IS,ID,IE,IL,IT,JP,LV,LT,LU,MY,MX,MD,NL,NZ,MK,NO,PL,PT,RO,RS,SG,SK,SI,ZA,KR,ES,SE,CH,TW,TH,TR,UA,AE,GB,US,VN, or Albania,Argentina,Australia,Austria,Belgium,Bosnia and Herzegovina,Brazil,Bulgaria,Canada,Chile,Colombia,Costa Rica,Croatia,Cyprus,Czech Republic,Denmark,Estonia,Finland,France,Georgia,Germany,Greece,Hong Kong,Hungary,Iceland,Indonesia,Ireland,Israel,Italy,Japan,Latvia,Lithuania,Luxembourg,Malaysia,Mexico,Moldova,Netherlands,New Zealand,North Macedonia,Norway,Poland,Portugal,Romania,Serbia,Singapore,Slovakia,Slovenia,South Africa,South Korea,Spain,Sweden,Switzerland,Taiwan,Thailand,Turkey,Ukraine,United Arab Emirates,United Kingdom,United States,Vietnam,. Please check https://haugene.github.io/docker-transmission-openvpn/provider-specific/#nordvpn
INFO: OVPN: Searching for group: legacy_p2p
INFO: OVPN:Searching for technology: openvpn_tcp

**Comment by edgd1er** [comment]
> adding route to local network 192.168.1.0/24 via 172.17.0.1 dev eth0
RTNETLINK answers: Operation not permitted

Your problem is here, when route are added.
You may add `DEBUG=true` to display commands and see exactly which command is the cause of the constant reboot.

what is your system ? Have you tried with /dev/tun as a device and CREATE_TUN as false ?

**Comment by SlippingEels** [comment]
Host is Ubuntu Server 22.04

I have set the Debug flag but it looks as though I have been banned from Nord for too many requests. Setting the CREATE_TUN_DEVICE as false now.

The container is being created and ran from Portainer.

**Comment by edgd1er** [comment]
Ubuntu being derived from Debian is known to be compatible with that container.
You'll have to wait a while for Nordvpn to allow you again to access the API. most of the time 1 hour is enough.

You may also add 'DROP_DEFAULT_ROUTE=false' just during the tests to validate the connection. I strongly advise to put it back to true whenever you are using the container.
so wait and see, for the moment.

**Comment by SlippingEels** [comment]
While I am waiting for my account to be unbanned, under Capabilities in Portainer, should I have NET_ADIN and SYS_MODULE enabled? I have just switched these both on.

**Comment by edgd1er** [comment]
[NET_ADMIN](https://docs.docker.com/engine/reference/run/) is required for network operations (create tun if not existing, route add/del, ...)
SYS_ADMIN as I recall, is only required when using dhcp.

**Comment by SlippingEels** [comment]
Thanks for you quick responses. 

It was indeed enabling NET_ADMIN which fixed the problem. This can be found in the Portainers Container advance options for those having the same issue.
